core/src/main/kotlin/no/strooped/StroopedController.kt

```diff
@@ -44,8 +46,17 @@ class StroopedController : Game() {
 
     fun connectToGame(username: String, joinPin: String) {
         joinGameService.joinGame(username, joinPin) {
-            GameSingleton.room = it
-            changeScreen(LobbyScreen(this))
+            it.fold(
+                { gameRoom ->
+                    GameSingleton.room = gameRoom
+                    changeScreen(LobbyScreen(this))
+                },
+                { err ->
+                    println("Failed to join game!: ${err.message}")
+                    // Re-render JoinGameScreen with error message
+                    changeScreen(JoinGameScreen(this, err.localizedMessage))
```

Comment:
Here we are rerendering the JoinGameScreen where we include a message which should be human readable.

Preferably should we also pass down the typed username and joinPin, so the user does not have to retype everything, but it might take more time than it is worth